core/genesis/genesis.go

```diff
@@ -0,0 +1,49 @@
+package genesis
+
+import (
+	"encoding/json"
+	"io"
+
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+// Source is a source for genesis data in JSON format. It may abstract over a
+// single JSON object or JSON in separate files that can be streamed over. Those
+// details are left to the implementation. Generally modules should either
+// consume a single JSON blob using the ReadMessage or ReadRawJSON methods OR
+// ideally, read individual fields using the OpenReader method. Using the
+// OpenReader method with array data will allow that data to be streamed which
+// can result in lower memory usage than reading a single large JSON blob into
+// memory.
+type Source interface {
+	// ReadRawJSON reads the all the genesis data as a single JSON blob.
+	ReadRawJSON() (json.RawMessage, error)
```

Comment:
json.RawMessage was needed because of the ORM, right? Do we still want to keep that now with collections?

Reply:
Ideally both of them use streaming